### DISCUSSION TITLE
Erreur de calcul ou d'affichage?

### DISCUSSION THREAD

**Discussion by abesnier** [bokub/conso-api#45] [Aide]
Bonjour, 

En regardant ma conso des jours passés, je vois que pour le ~~2~~ 3 avril, il y a un GROS pic de conso entre 2h et 3h du matin (10kWh!)
![image](https://github.com/bokub/conso-api/assets/19927690/33f4bff6-0f3a-4053-8185-26c152d314e5)

En regardant sur Enedis, rien d'anormal, la conso entre ces heures correspond bien à la baseline de ma conso (~0.2/0.3 kWh), à la fois sur les graphs de suivi ou en téléchargeant les données en CSV.
![image](https://github.com/bokub/conso-api/assets/19927690/8d551d13-c97f-4aa2-a965-be151a6db29d)

Ce pic semble correspondre à la somme de la conso de toute la journée.

Je ne sais pas si c'est un problème de l'api ou de ha-linky (qui tourne sous docker). 

Donc, comment identifier d'où vient le problème, et est-ce qu'il y a quelque chose que je peux faire pour corriger cela?

Cheers
Antoine

**Comment by bokub** [comment]
Peux-tu essayer de récupérer les donnés horaires du 3 avril sur la page examples de Conso API et me dire ce que tu obtiens ?

**Comment by abesnier** [comment]
Bonjour, 

Merci de la rapidité.
Depuis la page de test de l'api, les données sont bonnes et correspondent à ce que je vois chez Enedis:
![image](https://github.com/bokub/conso-api/assets/19927690/4fd123b9-11a1-4bd8-b1cf-10949edf0921)

Et l'image docker a été build il y a 11 jours, avec l'instruction sur le README.

**Comment by abesnier** [comment]
J'en rajoute une petite couche, les données dans la db homeassistant semblent également bonnes (données pour 2h, 3h et 4h du matin le 3 avril en unix epoch)
![image](https://github.com/bokub/conso-api/assets/19927690/36837fa1-de86-4e2f-b5b6-1891cb42908c)

l'ID 4 correspond à linky:prm, et les valeurs sont exactement celles d'enedis SAUF qu'il y a une heure de décalage. La machine host  et les containers étant à la bonne heure, je prendrai le temps de voir si cette erreur remonte à avant le changement d'heure accessoirement...)

**Comment by bokub** [comment]
@abesnier Si j'ai bien compris ton dernier message, les données dans la DB sont correctes, mais c'est uniquement le graphique qui montre n'importe quoi ?
Est-ce que tu as une colonne "sum" dans ta DB ?

**Comment by abesnier** [comment]
Bonjour, 

En effet, il y a un problème dans la colonne sum pour cette heure-ci:
![image](https://github.com/bokub/conso-api/assets/19927690/760c1d0d-3a82-4c76-947a-1034372537be)

10727 correspond à 3791806-3781079, et à la valeur du pic observé sur le graphe. (10,73kWh)
(l'heure est en UTC, donc +2 pour notre heure locale de France).

**Comment by bokub** [comment]
Alors là franchement je comprend pas, surtout en plein milieu d'une journée...

Si tu veux corriger tes données, il y a un outil dédié dans le menu "Outils de développement" de Home Assistant (`/developer-tools/statistics`), qui permet de rectifier une somme...

Sinon tu peux aussi remettre tes données à 0 et repartir d'un CSV propre

Je ne pense pas trouver la cause du souci, sauf si tu arrives à reproduire la même erreur

**Comment by abesnier** [comment]
J'ai corrigé avec les outils de développement.
Ce qui est étonnant, c'est que la valeur 10727 ne correspond à rien, j'ai essayé de sommer les valeurs sur 24h, 23h, 22h, rien ne donne 10727...
Je vais garder un oeil, voir si ça se reproduit. En attendant, on va mettre ça sur une erreur de calcul de JavaScript (ha il a bon dos JS!)
Merci quand même, et merci pour l'outil
